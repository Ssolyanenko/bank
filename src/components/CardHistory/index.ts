export * from './History';
export * from './Pagination';
export * from './CardHistory';
