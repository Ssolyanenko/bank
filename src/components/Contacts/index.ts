export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
