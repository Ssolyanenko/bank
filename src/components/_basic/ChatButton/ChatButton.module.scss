.chatButton {
  position: absolute;
  right: 2rem;
  bottom: 14.5rem;
  cursor: pointer;
  border-radius: 50%;
}
