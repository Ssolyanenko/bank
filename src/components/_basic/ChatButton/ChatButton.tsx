import { FC, ReactElement } from 'react';

import { ChatIcon } from 'assets';
import { toDo } from 'helpers';
import { ButtonType } from 'interfaces/common/componentsSettings';
import classes from './ChatButton.module.scss';

interface Props {
  name?: string;
  type?: ButtonType;
  onClick?(): void;
}

export const ChatButton: FC<Props> = ({ type = ButtonType.BUTTON, name = '', onClick = toDo }): ReactElement => (
  <button type={type} name={name} onClick={onClick}>
    <ChatIcon className={classes.chatButton} />
  </button>
);
