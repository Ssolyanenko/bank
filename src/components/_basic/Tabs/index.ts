export * from './Tabs';
export * from './TabTitle';
