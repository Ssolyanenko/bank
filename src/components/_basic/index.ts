export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
