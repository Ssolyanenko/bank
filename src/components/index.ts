export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
