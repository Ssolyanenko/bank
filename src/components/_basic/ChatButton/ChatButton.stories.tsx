import { ComponentMeta, ComponentStory } from '@storybook/react';

import { ChatButton } from './ChatButton';

export default {
  title: 'Basic/Buttons/ChatButton',
  component: ChatButton,
  parameters: {
    docs: {
      description: {
        component: 'Button to open support chat',
      },
    },
  },
} as ComponentMeta<typeof ChatButton>;

const Template: ComponentStory<typeof ChatButton> = (props) => <ChatButton {...props} />;

export const Default = Template.bind({});

Default.args = {
  onClick(): void {
    alert('Click open chat button');
  },
};
