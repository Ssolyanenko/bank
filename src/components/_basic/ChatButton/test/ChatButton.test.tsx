import { shallow } from 'enzyme';

import { ChatButton } from 'components/_basic';
import { Default } from 'components/_basic/ChatButton/ChatButton.stories';

describe('<ChatButton />', (): void => {
  test('should make a snapshot of component', (): void => {
    const wrapper = shallow(<ChatButton />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should render ChatButton from storybook', (): void => {
    const wrapper = shallow(<Default />);

    expect(wrapper.find(ChatButton).length).toBe(1);
  });
});
