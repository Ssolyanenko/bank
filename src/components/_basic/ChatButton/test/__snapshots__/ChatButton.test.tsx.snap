// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatButton /> should make a snapshot of component 1`] = `
<button
  name=""
  onClick={[Function]}
  type="button"
>
  <ChatIcon
    className="chatButton"
  />
</button>
`;
